src/uu/w/src/w.rs

```diff
@@ -83,6 +83,54 @@ fn fetch_pcpu_time(pid: i32) -> Result<f64, std::io::Error> {
     Ok((utime + stime) / get_clock_tick() as f64)
 }
 
+#[cfg(target_os = "linux")]
+fn fetch_idle_time(tty: String) -> Result<Duration, std::io::Error> {
+    let path = Path::new("/dev/").join(tty);
+    let stat = fs::metadata(path)?;
+    if let Ok(time) = stat.accessed() {
+        Ok(SystemTime::now().duration_since(time).unwrap_or_default())
+    } else {
+        Ok(Duration::ZERO)
+    }
+}
+
+#[cfg(not(target_os = "linux"))]
+fn _fetch_idle_time(_tty: String) -> Result<Duration, std::io::Error> {
+    Ok(Duration::ZERO)
+}
+
+fn format_time_elapsed(
+    time: Duration,
+    old_style: bool,
+) -> Result<String, chrono::format::ParseError> {
```

Comment:
I'm not sure about this return type. The current implementation of the function never returns an `Err` and so the return type doesn't have to be a `Result`. On the other hand, it might make sense to use a `?` instead of `unwrap()` on line 106 and to adapt the error type accordingly.

Reply:
My intention was to comply with what we had earlier with `format_time`, which is also of this return type. Though in fact the `Result` on line 106 seems to have type of `chrono::OutOfRangeError`. I'll adapt that type instead.